### PR TITLE
[Benchmark] Add weekly scheduling for nightly benchmark

### PR DESCRIPTION
**Pull request by Linchenn**
This PR adds a flag `--weeklyRun` to run a part of models specified in the `preconfigured_browser.json` everyday and set week as the period.

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6959)

### CODE REVIEW

File: e2e/benchmarks/browserstack-benchmark/app.js [hunk 1]
```diff
@@ -362,6 +363,12 @@ function setupHelpMessage() {
     help: 'runs GCP compatible version of benchmarking system',
     action: 'store_true'
   });
+  parser.add_argument('--weeklyRun', {
+    help: 'runs a part of models specified in the file everyday (the period ' +
+      'is a week), if --benchmarks is set',
+    default: false,
+    action: 'store_true'
+  });
```
Comment (mattsoulanille) [inline e2e/benchmarks/browserstack-benchmark/app.js:371]: Nit: It might be nice to have a way to run a specific day's tests for debugging purposes. Maybe if you pass a day of the week to this flag, it can run that day's tests, but by default, it uses the current date.

File: e2e/benchmarks/browserstack-benchmark/app_node_test.js [hunk 2]
```diff
@@ -313,20 +313,32 @@ describe('promise queue', () => {
...
+
+describe('weekly schedule', () => {
+  it('scheduling models works', () => {
+    const dt = new Date("October 19, 2022");  // Wednesday.
+    models = Array.from(Array(24).keys());
+    const res = scheduleModels(models, dt);
+    expect(res).toEqual(
+      [12, 13, 14, 15]
     );
   });
```
Comment (mattsoulanille) [inline e2e/benchmarks/browserstack-benchmark/app_node_test.js:343]: Nit: Maybe add some tests for edge cases like day 0 (Sunday) and day 6 (Saturday), although I'm pretty sure your function is correct.

File: e2e/benchmarks/browserstack-benchmark/app.js [hunk 3]
```diff
@@ -362,6 +363,12 @@ function setupHelpMessage() {
     help: 'runs GCP compatible version of benchmarking system',
     action: 'store_true'
   });
+  parser.add_argument('--weeklyRun', {
```
Comment (mattsoulanille) [inline e2e/benchmarks/browserstack-benchmark/app.js:366]: Nit: `--weeklyRun` implies it runs once every week.

```suggestion
  parser.add_argument('--dailyRun', {
```

File: e2e/benchmarks/browserstack-benchmark/app.js [hunk 4]
```diff
@@ -409,7 +429,11 @@ function setupHelpMessage() {
  */
 async function runBenchmarkFromFile(file, runBenchmark = benchmarkAll) {
   console.log('Running a preconfigured benchmark...');
-  await runBenchmark(file);
+  const { benchmark, browsers } = file;
+  if (cliArgs.weeklyRun) {
```
Comment (mattsoulanille) [inline e2e/benchmarks/browserstack-benchmark/app.js:433]:
```suggestion
  if (cliArgs.dailyRun) {
```

File: e2e/benchmarks/browserstack-benchmark/cloudbuild.yml [hunk 5]
```diff
@@ -12,7 +12,7 @@ steps:
   dir: 'e2e/benchmarks/browserstack-benchmark'
   entrypoint: 'yarn'
   id: 'test'
-  args: ['run-cloud-benchmarks']
+  args: ['run-cloud-benchmarks', '--weeklyRun']
```
Comment (mattsoulanille) [inline e2e/benchmarks/browserstack-benchmark/cloudbuild.yml:15]:
```suggestion
  args: ['run-cloud-benchmarks', '--dailyRun']
```

**Review by mattsoulanille — APPROVED** [review]
Looks good, I just have a few nits. Thanks!

File: e2e/benchmarks/browserstack-benchmark/app.js (hunk 3 above)
Comment (Linchenn) [reply e2e/benchmarks/browserstack-benchmark/app.js:366]: Good suggestion. What about 'weeklyCycleRun'? In addition to distinguishing from 'weekly run', it indicates 'weekly cycle'?

File: e2e/benchmarks/browserstack-benchmark/app_node_test.js (hunk 2 above)
Comment (Linchenn) [reply e2e/benchmarks/browserstack-benchmark/app_node_test.js]: Done.

File: e2e/benchmarks/browserstack-benchmark/app.js (hunk 3 above)
Comment (mattsoulanille) [reply e2e/benchmarks/browserstack-benchmark/app.js:366]: Sounds good to me.

Comment (Linchenn) [reply e2e/benchmarks/browserstack-benchmark/app.js]: Thank you!

File: e2e/benchmarks/browserstack-benchmark/app.js (hunk 1 above)
Comment (Linchenn) [reply e2e/benchmarks/browserstack-benchmark/app.js]: Great catch, done. Thank you!

**Comment by Linchenn** [comment]
Added Na in case Ping is OOO recently.

**Review by lina128 — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6959#-:-NEr2oes15edUUo0yQdC:bb6ywmx)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @Linchenn, @mattsoulanille, and @pyu10055)
